### PR TITLE
Updating SQL vnet rules and tests with migration flag and state

### PR DESCRIPTION
**Pull request by kisantia**
## Description

- Adding state to output of New, Set, and Get Vnet cmdlets
- Adding migration flag to New and Set Vnet cmdlets as a switch parameter (-IgnoreMissingVnetServiceEndpoint)
- These changes will be in Microsoft.Azure.Management.Sql 1.8.0-preview (https://github.com/Azure/azure-sdk-for-net/pull/3740), but the SDK is not yet published so the build will fail

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [ ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by cormacpayne** [comment]
@azuresdkci add to whitelist

**Comment by markcowl** [comment]
@kisantia please fix merge conflicts and fix any ci build issues

**Comment by kisantia** [comment]
@markcowl I fixed the merge conflicts, but the build won't pass until Microsoft.Azure.Management.Sql.1.8.0-preview is published hopefully sometime this week

**Comment by jaredmoo** [comment]
@markcowl @shahabhijeet we need https://github.com/Azure/azure-sdk-for-net/pull/3800 to be merged and then we will publish Microsoft.Azure.Management.Sql.1.8.0-preview

File: src/ResourceManager/Sql/Commands.Sql/help/New-AzureRmSqlServerVirtualNetworkRule.md [hunk 1]
```diff
@@ -29,6 +32,34 @@ Creates an Azure SQL Server virtual network rule
 
 ## PARAMETERS
 
+### -DefaultProfile
+The credentials, account, tenant, and subscription used for communication with azure.```yaml
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/help/New-AzureRmSqlServerVirtualNetworkRule.md]: @kisantia this is a bug in `platyPS` - make sure that the ` ```yaml  string is on a line by itself. See the other parameters in this file as guidance.

File: src/ResourceManager/Sql/Commands.Sql/help/Remove-AzureRmSqlServerVirtualNetworkRule.md [hunk 2]
```diff
@@ -30,13 +32,11 @@ Deletes an existing Azure SQL Server virtual network rule
 
 ## PARAMETERS
 
-### -Force
-Skip confirmation message for performing the action
-
-```yaml
-Type: SwitchParameter
+### -DefaultProfile
+The credentials, account, tenant, and subscription used for communication with azure.```yaml
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/help/Remove-AzureRmSqlServerVirtualNetworkRule.md]: @kisantia same comment

File: src/ResourceManager/Sql/Commands.Sql/help/Set-AzureRmSqlServerVirtualNetworkRule.md [hunk 3]
```diff
@@ -33,6 +35,34 @@ Modifies an existing virtual network rule with the new virtual network subnet id
 
 ## PARAMETERS
 
+### -DefaultProfile
+The credentials, account, tenant, and subscription used for communication with azure.```yaml
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/help/Set-AzureRmSqlServerVirtualNetworkRule.md]: @kisantia same comment

**Comment by markcowl** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/65/

**Pull request merged**